test/integration/tcp_proxy_integration_test.h

```diff
@@ -25,9 +26,24 @@ class TcpProxyIntegrationTest : public BaseIntegrationTest,
     test_server_.reset();
     fake_upstreams_.clear();
   }
+};
 
+class TcpProxySslIntegrationTest : public TcpProxyIntegrationTest {
```

Comment:
Would it make more sense to have tls be an argument to initialize_ and call most/all tests with a TEST_P?  Or generalize the tests in the base class as we do in http_integration.cc and call them from TEST_F(Tls...) TEST_F(Raw) like we do for H2/HTTP/H2Upstream?

Reply:
I like going in the direction of making TEST_P's for the tests.  However, the TLS tests end up being written slightly differently: they require a bunch of calls to dispatcher_->run(), I think because of the behavior of the ssl library.  I'm not entirely sure why that is needed, and I'd rather not sort out that knot in this PR.  It's already big enough.

Reply:
Fair enough - you've gone great testing so I'll only ask you do so much rewriting :-P  I'll keep it on my list for test framework improvements then!